File: app/src/main/java/app/quranhub/ui/mushaf/data/dao/TranslationDao.java
package app.quranhub.ui.mushaf.data.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

import app.quranhub.ui.mushaf.data.entity.Translation;
import io.reactivex.Single;

@Dao
public interface TranslationDao {

    @Query("SELECT * FROM translation")
    Single<List<Translation>> getAll();

    @Query("SELECT * FROM translation where `sura`=:suraNumber")
    LiveData<List<Translation>> getAyasTafseer(int suraNumber);

    @Query("SELECT text FROM translation WHERE `index`=:index")
    Single<String> findByIndex(int index);

    @Query("SELECT * FROM translation WHERE sura=:sura AND aya=:aya LIMIT 1")
    Translation findForAya(int sura, int aya);

}

File: app/src/main/java/app/quranhub/data/local/dao/TranslationDao.kt
package app.quranhub.data.local.dao

import androidx.room.Dao
import androidx.room.Query
import app.quranhub.data.local.entity.Translation
import kotlinx.coroutines.flow.Flow
import io.reactivex.Single

@Dao
interface TranslationDao {

    @Query("SELECT * FROM translation")
    fun getAll(): Single<List<Translation>>

    @Query("SELECT * FROM translation where `sura`=:suraNumber")
    fun getAyasTafseer(suraNumber: Int): Flow<List<Translation>>

    @Query("SELECT text FROM translation WHERE `index`=:index")
    fun findByIndex(index: Int): Single<String>

    @Query("SELECT * FROM translation WHERE sura=:sura AND aya=:aya LIMIT 1")
    fun findForAya(sura: Int, aya: Int): Translation?
}